src/Raven.Server/Documents/ShardedTcpHandlers/ShardedSubscriptionWorker.cs

```diff
@@ -0,0 +1,112 @@
+﻿using System;
+using System.Diagnostics;
+using System.IO;
+using System.Security.Cryptography.X509Certificates;
+using System.Threading.Tasks;
+using Raven.Client.Documents.Conventions;
+using Raven.Client.Documents.Subscriptions;
+using Raven.Client.Http;
+using Raven.Client.ServerWide.Tcp;
+using Sparrow.Json;
+
+namespace Raven.Server.Documents.ShardedTcpHandlers
+{
+    public class ShardedSubscriptionWorker : AbstractSubscriptionWorker<dynamic>
+    {
+        private readonly RequestExecutor _shardRequestExecutor;
+
+        public ShardedSubscriptionWorker(SubscriptionWorkerOptions options, string dbName, RequestExecutor re) : base(options, dbName)
+        {
+            _shardRequestExecutor = re; 
+        }
+
+        internal override RequestExecutor GetRequestExecutor()
+        {
+            return _shardRequestExecutor;
+        }
+
+        internal override void GetLocalRequestExecutor(string url, X509Certificate2 cert)
+        {
+            _subscriptionLocalRequestExecutor?.Dispose();
+            _subscriptionLocalRequestExecutor = RequestExecutor.CreateForSingleNodeWithoutConfigurationUpdates(url, _dbName, cert, DocumentConventions.Default);
+        }
+
+        internal override bool ShouldUseCompression()
+        {
+            bool compressionSupport = false;
+#if NETCOREAPP3_1_OR_GREATER
+            var version = SubscriptionTcpVersion ?? TcpConnectionHeaderMessage.SubscriptionTcpVersion;
+            if (version >= 53_000 && (_shardRequestExecutor.Conventions.DisableTcpCompression == false))
+                compressionSupport = true;
+#endif
+            return compressionSupport;
+        }
+
+        public class PublishedBatch
+        {
+            internal BatchFromServer _batchFromServer;
+            public TaskCompletionSource SendBatchToClientTcs;
+            public TaskCompletionSource ConfirmFromShardSubscriptionConnectionTcs;
+            public string LastSentChangeVectorInBatch;
+        }
+
+        public PublishedBatch PublishedBatchItem;
+
+        internal override async Task ProcessSubscriptionInternal(JsonContextPool contextPool, Stream tcpStreamCopy, JsonOperationContext.MemoryBuffer buffer, JsonOperationContext context)
```

Comment:
We need a place with a comment that explains the interactions between client, orchestrator and shards

Reply:
done